lib/matplotlib/tests/test_backends_interactive.py

```diff
@@ -319,3 +318,47 @@ def test_lazy_linux_headless():
     if proc.returncode:
         pytest.fail("The subprocess returned with non-zero exit status "
                     f"{proc.returncode}.")
+
+
+# The source of this function gets extracted and run in another process, so it
+# must be fully self-contained.
+def _test_figure_leak():
+    import json
+    import sys
+
+    import psutil
+    from matplotlib import pyplot as plt, rcParams
+
+    rcParams.update({
+        "webagg.open_in_browser": False,
+        "webagg.port_retries": 1,
+    })
+    if len(sys.argv) >= 2:  # Second argument is json-encoded rcParams.
+        rcParams.update(json.loads(sys.argv[1]))
+
+    # We can't yet directly identify the leak
+    # so test with a memory growth threshold
+    ACCEPTABLE_MEMORY_LEAKAGE = 1_000_000
+
+    # Warmup cycle, this reasonably allocates a lot
+    p = psutil.Process()
+    fig = plt.figure()
+    plt.close(fig)
+    mem = p.memory_full_info().uss
+
+    # test cycle, this seems robust enough but
+    # consider looping or averaging in the future
+    fig = plt.figure()
+    plt.close(fig)
+    growth = p.memory_full_info().uss - mem
+    assert growth < ACCEPTABLE_MEMORY_LEAKAGE, "Memory leakage too high"
```

Comment:
Since this is failing, probably want to mention by how much (this isn't run in pytest, so won't get any argument parsing).